File: src/main/java/com/dgf/casumotest/repo/CustomerRepo.java
package com.dgf.casumotest.repo;

import com.dgf.casumotest.model.Customer;
import org.springframework.data.mongodb.repository.ReactiveMongoRepository;
import reactor.core.publisher.Flux;


public interface CustomerRepo extends ReactiveMongoRepository<Customer, String> {

    Flux<Customer> findByFirstNameAndLastNameIgnoreCase(String firstName, String lastName);

}

File: src/main/kotlin/com/dgf/casumotest/repo/CustomerRepo.kt
package com.dgf.casumotest.repo

import com.dgf.casumotest.model.Customer
import org.springframework.data.mongodb.repository.ReactiveMongoRepository
import reactor.core.publisher.Flux


interface CustomerRepo : ReactiveMongoRepository<Customer, String> {

    fun findByFirstNameAndLastNameIgnoreCase(firstName: String, lastName: String): Flux<Customer>

}
